hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMHAManagerImpl.java

```diff
@@ -355,6 +356,26 @@ public boolean addSCM(AddSCMRequest request) throws IOException {
     return getRatisServer().addSCM(request);
   }
 
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public boolean removeSCM(RemoveSCMRequest request) throws IOException {
+    // Currently we don't support decommission of Primordial SCM
+    // The caller should make sure that the requested node is not Primordial SCM
+
+    String clusterId = scm.getClusterId();
+    if (!request.getClusterId().equals(scm.getClusterId())) {
+      throw new IOException("SCM " + request.getScmId() +
```

Comment:
Thanks @nandakumar131 for providing support for removing an SCM from the ratis ring with this patch.
For clarity, would be helpful to add cause message for exception at start of `IOException `message, `"Cannot remove primordial node from ring.  SCM " + request.getScmId() + ...`

Reply:
@neils-dev, the check that is done here is to make sure that the request has the correct cluster-ID and matches the current SCM's cluster-ID.

The primordial check has to be done by the caller of this method. This method assumes that the current SCM is not primordial SCM.

Primordial check is done in `StorageContainerManager#removePeerFromHARing`